src/server/handlers.rs

```diff
@@ -762,3 +747,158 @@ impl<P: Publisher> Handler<P> {
         self.send_reply(reply, response).await;
     }
 }
+
+pub async fn open_schema_file() -> anyhow::Result<BufReader<File>> {
+    let schema_file = File::open("./oam/oam.schema.json")?;
+    Ok(BufReader::new(schema_file))
+}
+
+// Manifest validation
+pub(crate) fn validate_manifest(
+    manifest: Manifest,
+    current_manifests: StoredManifest,
+) -> Option<String> {
```

Comment:
nit: I don't find this too bad, but I think it could be more clear here to use a `Result<()>` and return an error if the validation fails. That way, an assertion like `validate_manifest().is_ok()` would mean it's valid, over `validate_manifest().is_none()`

Not hard set on this and if you disagree, a doc comment indicating the purpose of the return type would be enough.